### ISSUE TITLE
--disable-web-security is not working 

### ISSUE THREAD

**Issue by alanEG** [microsoft/playwright#27373]
### System info
- Playwright Version: 1.38.0
- Operating System: Ubuntu 20.04.4 LTS (Focal Fossa) (WSL)
- Browser: WebKit
- Python Version: 3.9

### Source code

```python
async def get_detail(context, url):
    page = await context.new_page()
    try:
        await page.route(url,handle_route)
        await handle_goto(page,url) 
        await page.wait_for_load_state(state="networkidle")
        await page.wait_for_timeout(1000)        
        check_content(await page.content(),url)
        # For debugging purposes
        time.sleep(100)
        await page.close()
    except Exception as e:
        print("<<", url, e)
        await page.close()
 
    
async def open_new_pages(context, urls):
    semaphore = asyncio.Semaphore(1)
    async def process_url(url):
        async with semaphore:
            await get_detail(context,url)
    
    tasks = []
    for url in urls:
        task = asyncio.create_task(process_url(url))
        tasks.append(task)
    
    await asyncio.gather(*tasks)

async def main():
    async with async_playwright() as p:
        browser = await p.webkit.launch(headless=False,args=["--disable-web-security=TRUE"])
        context = await browser.new_context()
        urls = []
        for line in sys.stdin:
            urls.append(line.strip("\ufeff").rstrip())
       
        await open_new_pages(context, urls)
    
asyncio.run(main())
```
**Steps**
- [Run the test]

**Expected**

![image](https://github.com/microsoft/playwright/assets/52584554/55d65d9d-c10b-4dc5-92d9-98bde0b84352)

**Actual**

![image](https://github.com/microsoft/playwright/assets/52584554/276bcb33-da50-4422-ac2b-4170dbfc6a72)

**Comment by yury-s** [comment]
There is no such command line option in WebKit, what are you trying to achieve?

**Comment by alanEG** [comment]
Hi @yury-s 
I found https://github.com/microsoft/playwright/issues/15757, so I assumed that the option would be available for all browsers.

I am currently facing some problems like CSP and mixed content issues, so I want to disable the security features in WebKit.

**Comment by yury-s** [comment]
Have you tried [`bypass_csp`](https://playwright.dev/python/docs/api/class-browser#browser-new-context-option-bypass-csp) option?

**Comment by alanEG** [comment]
I have tested it, and here are the browser console logs.

Without the option `bypass_csp`
```
[Error] Refused to connect to https://localhost/test.txt because it does not appear in the connect-src directive of the Content Security Policy.
```

With the option `bypass_csp`
```
[Error] Failed to load resource: Blocked by Content Security Policy. (test.txt, line 0)
[Error] Blocked by Content Security Policy.
[Error] XMLHttpRequest cannot load https://localhost/test.txt due to access control checks.
[Error] NetworkError:  A network error occurred.
[Error] Refused to connect to https://localhost/test.txt because it does not appear in the connect-src directive of the Content Security Policy.
```

**Comment by pavelfeldman** [comment]
If you would like to disable the CSP, have tried using the `bypass_csp` option and it did not work, please share the repro scenario and we will be able to act on it.

**Comment by alanEG** [comment]
Hello, @pavelfeldman,

I'm making some edits to the response before the page loads, and I'm also including some JavaScript. This JavaScript is attempting to fetch content from a local HTTP/HTTPS server. However, due to Content Security Policy (CSP) and other security features, I'm unable to load content from the local server.

Code:
```python
start_script = """
<script>
  function loadWordlist(url){
    var xhttp = new XMLHttpRequest();
    xhttp.onreadystatechange = function() {
        if (this.readyState == 4 && this.status == 200) {
          output=run(xhttp.responseText.split('\\n'));
          window.history.pushState({},"",output);
        }
    };
    xhttp.open("GET", url, false);
    xhttp.send();
  }
  loadWordlist('{wordlist_url}');
</script>
"""
async def modify_response(response,headers,url):
    if '<head' in response:
        match_html = re.findall('(<head.*?>)',response)[0]
        response = response.replace(match_html,match_html + start_script)
    elif '<body' in response:
        match_html = re.findall('(<body.*?>)',response)[0]
        response= response.replace(match_html,match_html + start_script)
    elif 'Content-Type' in headers and 'text/html' in headers['Content-Type']:
        response =  start_script + response

    return response.replace('{wordlist_url}','https://127.0.0.1/wordlist.txt')

async def handle_route(route) -> None:
    # Fetch response.
    try:
        response = await route.fetch()
        body = await response.text()
        body = await modify_response(body,response.headers,response.url)
        await route.fulfill(
            response=response,
            body=body,
        )
    except Exception as e:
        print(e)
        # This try because the response.text() will reaise an exception if the content is image or binary
        await route.continue_()
        
async def get_detail(context, url):
    page = await context.new_page()
    try:
        await page.route(url,handle_route)
        await handle_goto(page,url) 
        await page.wait_for_load_state(state="networkidle")
        await page.wait_for_timeout(1000)        
        check_content(await page.content(),url)
        # For debugging purposes
        time.sleep(100)
        await page.close()
    except Exception as e:
        print("<<", url, e)
        await page.close()
 
    
async def open_new_pages(context, urls):
    semaphore = asyncio.Semaphore(1)
    async def process_url(url):
        async with semaphore:
            await get_detail(context,url)
    
    tasks = []
    for url in urls:
        task = asyncio.create_task(process_url(url))
        tasks.append(task)
    
    await asyncio.gather(*tasks)

async def main():
    async with async_playwright() as p:
        browser = await p.webkit.launch(headless=False,args=[""])
        context = await browser.new_context()
        urls = []
        for line in sys.stdin:
            urls.append(line.strip("\ufeff").rstrip())
       
        await open_new_pages(context, urls)
    
asyncio.run(main())
```

**Comment by pavelfeldman** [comment]
I'm sorry, we can only act on something we can repro, it is hard to guess what the steps are, what the expected and actual outcomes are. Please file a new issue and use the Bug template to provide all the details.
